app/v2/controllers/V2MovementsController.scala

```diff
@@ -0,0 +1,328 @@
+  val preMaterialisedFutureProvider: PreMaterialisedFutureProvider
+)(implicit val materializer: Materializer, val temporaryFileCreator: TemporaryFileCreator)
+    extends BaseController
+    with V2MovementsController
+    with Logging
+    with StreamingParsers
+    with VersionedRouting
+    with ErrorTranslator
+    with ContentTypeRouting
+    with HasActionMetrics {
+
+  lazy val sCounter: Counter = counter(s"success-counter")
+  lazy val fCounter: Counter = counter(s"failure-counter")
+
+  def createMovement(movementType: MovementType): Action[Source[ByteString, _]] =
+    movementType match {
+      case MovementType.Arrival =>
+        contentTypeRoute {
+          case Some(MimeTypes.XML)  => submitArrivalNotificationXML()
+          case Some(MimeTypes.JSON) => submitArrivalNotificationJSON()
+        }
+      case MovementType.Departure =>
+        contentTypeRoute {
+          case Some(MimeTypes.XML)  => submitDepartureDeclarationXML()
+          case Some(MimeTypes.JSON) => submitDepartureDeclarationJSON()
+        }
+    }
+
+  private def submitDepartureDeclarationXML(): Action[Source[ByteString, _]] =
+    (authActionNewEnrolmentOnly andThen messageSizeAction()).stream {
+      implicit request =>
+        implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequest(request)
+
+        (for {
+          _ <- validationService.validateXml(MessageType.DeclarationData, request.body).asPresentation
+          _ = auditService.audit(AuditType.DeclarationData, request.body, MimeTypes.XML)
+          movementResponse <- movementsService.createMovement(request.eoriNumber, MovementType.Departure, request.body).asPresentation
+          _ = pushNotificationsService.associate(movementResponse.movementId, MovementType.Departure, request.headers)
+          _ <- routerService
+            .send(MessageType.DeclarationData, request.eoriNumber, movementResponse.movementId, movementResponse.messageId, request.body)
+            .asPresentation
+        } yield movementResponse).fold[Result](
+          presentationError => Status(presentationError.code.statusCode)(Json.toJson(presentationError)),
+          response => Accepted(HateoasNewMovementResponse(response.movementId, MovementType.Departure))
+        )
+    }
+
+  private def submitDepartureDeclarationJSON(): Action[Source[ByteString, _]] =
+    (authActionNewEnrolmentOnly andThen messageSizeAction()).stream {
+      implicit request =>
+        implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequest(request)
+
+        (for {
+          _ <- validationService.validateJson(MessageType.DeclarationData, request.body).asPresentation
+          _ = auditService.audit(AuditType.DeclarationData, request.body, MimeTypes.JSON)
+          xmlSource         <- conversionService.jsonToXml(MessageType.DeclarationData, request.body).asPresentation
+          declarationResult <- persistAndSendToEIS(xmlSource, MovementType.Departure, MessageType.DeclarationData)
```

Comment:
I don't quite know why this has happened, but I thought the whole idea of `persistAndSendToEIS` was to unify the XML and Json routes -- but that's not happened here (it was the same when I was looking at the attach routes, it caught me out as I thought I only had to update something in one place).

Can we look to see if we can make the XML route also use this method, so that we don't have the implementation multiple times when we don't need to?